### PR TITLE
Enhance/5538 - Render TwG supporter wall notification if the module is available - fix follow up

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #5538 

## Relevant technical choices

- This follow-up PR fixes the issue caught in QA. Please refer to this [comment](https://github.com/google/site-kit-wp/issues/5538#issuecomment-1263322636).
- Basically, the fix is to render the TwG supporter wall notification if the module is available.
- Fixes the flickering notification issue found in this [comment](https://github.com/google/site-kit-wp/issues/5538#issuecomment-1262342041) by preloading the API call.
- Fixes the following React warning that doesn't appear when the notification is displayed.

Screenshot 2022-10-04 at 9 49 13 PM

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Comment by asvinb** [comment]
LGTM

File: assets/js/components/notifications/BannerNotifications.js [hunk 1]
```diff
@@ -56,6 +56,9 @@ export default function BannerNotifications() {
 	const adSenseModuleActive = useSelect( ( select ) =>
 		select( CORE_MODULES ).isModuleActive( 'adsense' )
 	);
+	const twGModuleAvailable = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleAvailable( 'thank-with-google' )
```
Comment (aaemnnosttv) [inline assets/js/components/notifications/BannerNotifications.js:60]: The value here should be the same as the state of the feature flag, unless someone of course filtered the module out. In that case, `isModuleAvailable` should be used in place of the `twg` feature flag for module-availability checks.

File: assets/js/components/notifications/BannerNotifications.js [hunk 2]
```diff
@@ -70,7 +73,7 @@ export default function BannerNotifications() {
 					{ isAuthenticated && <CoreSiteBannerNotifications /> }
 					{ dashboardSharingEnabled && <ModuleRecoveryAlert /> }
 					{ ga4ActivationBannerEnabled && <ActivationBanner /> }
-					{ twgEnabled && (
+					{ twgEnabled && twGModuleAvailable && (
```
Comment (aaemnnosttv) [inline assets/js/components/notifications/BannerNotifications.js:76]: This notification should only be shown when TwG is connected. As it is, it's triggering a request for the `supporter-wall-prompt` on the dashboard any time the feature is enabled, regardless of the state of the module which is not what we want.

Since a module cannot be connected without also being active and available, this is essentially the only condition we need here.

File: assets/js/components/notifications/ThankWithGoogleSupporterWallNotification.js [hunk 3]
```diff
@@ -47,24 +47,25 @@ export default function ThankWithGoogleSupporterWallNotification() {
...
 		<BannerNotification
 			id="twg-supporter-wall-prompt"
+			className="googlesitekit-twg-supporter-wall-banner"
 			title={ __(
 				'Add a Thank with Google supporter wall',
 				'google-site-kit'
 			) }
 			description={
 				<Fragment>
-					<p>
```
Comment (aaemnnosttv) [inline assets/js/components/notifications/ThankWithGoogleSupporterWallNotification.js:56]: Why was this changed to use a `span` in place of a `p` tag? Paragraph would be more appropriate for this case.

File: assets/sass/components/thank-with-google/_googlesitekit-twg-supporter-wall.scss [hunk 4]
```diff
@@ -41,4 +41,11 @@
 			margin-top: 30px;
 		}
 	}
+
+	.googlesitekit-twg-supporter-wall-banner {
+
+		.googlesitekit-twg-supporter-wall-banner__description {
+			display: block;
+		}
```
Comment (aaemnnosttv) [inline assets/sass/components/thank-with-google/_googlesitekit-twg-supporter-wall.scss:49]: This was added due to the change to a `span` tag correct? I don't think it's needed otherwise (see question above). If it really was needed, it would be better to use the `.googlesitekit-display-block` utility class instead.

File: includes/Modules/Thank_With_Google.php [hunk 5]
```diff
@@ -133,6 +134,18 @@ function() {
 			}
 		);
 
+		add_filter(
+			'googlesitekit_apifetch_preload_paths',
+			function ( $paths ) {
+				return array_merge(
+					$paths,
+					array(
+						'/' . REST_Routes::REST_ROOT . '/modules/thank-with-google/data/supporter-wall-prompt',
```
Comment (aaemnnosttv) [inline includes/Modules/Thank_With_Google.php:143]: This request should technically work if TwG is active, but there isn't much point if the module isn't connected. Let's mirror the logic for showing the banner to only preload this if connected (we can just move this after the relevant guard just below this).

File: includes/Modules/Thank_With_Google.php [hunk 6]
```diff
@@ -410,9 +423,9 @@ function ( $host ) {
 					$is_connected            = $this->is_connected();
 					$setup_transient         = $this->transients->get( self::TRANSIENT_SETUP_TIMER );
 					if ( $is_connected && empty( $supporter_wall_sidebars ) && ! $setup_transient ) {
```
Comment (aaemnnosttv) [inline includes/Modules/Thank_With_Google.php:425]: The `is_connected` state doesn't really belong here – we shouldn't be calling this endpoint before the module is connected.

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @hussain-t – this needs a few changes before we can merge it. See my comments below and let me know if you have any questions 👍

File: assets/js/components/notifications/ThankWithGoogleSupporterWallNotification.js (hunk 3 above)
Comment (hussain-t) [reply assets/js/components/notifications/ThankWithGoogleSupporterWallNotification.js:56]: I have changed it due to the React warning that I mentioned in the PR description.

Screenshot 2022-10-04 at 9 49 13 PM

The error is caused since the `description` is rendered within the `<p>` tag in the `BannerNotification`. However, the warning appears when we try to nest the strings with the `<p>` tag. Hope that makes sense.

File: assets/sass/components/thank-with-google/_googlesitekit-twg-supporter-wall.scss (hunk 4 above)
Comment (hussain-t) [reply assets/sass/components/thank-with-google/_googlesitekit-twg-supporter-wall.scss:49]: Sure 👍

File: assets/js/components/notifications/BannerNotifications.js (hunk 2 above)
Comment (hussain-t) [reply assets/js/components/notifications/BannerNotifications.js]: Updated.

File: assets/sass/components/thank-with-google/_googlesitekit-twg-supporter-wall.scss (hunk 4 above)
Comment (hussain-t) [reply assets/sass/components/thank-with-google/_googlesitekit-twg-supporter-wall.scss]: Updated.

File: includes/Modules/Thank_With_Google.php (hunk 6 above)
Comment (hussain-t) [reply includes/Modules/Thank_With_Google.php]: Updated.

**Comment by hussain-t** [comment]
Thanks, @aaemnnosttv. I have addressed the comments.

File: assets/js/components/notifications/ThankWithGoogleSupporterWallNotification.js (hunk 3 above)
Comment (aaemnnosttv) [reply assets/js/components/notifications/ThankWithGoogleSupporterWallNotification.js:56]: Ah I see. I don't think we should be rendering the `description` in a wrapping `p` tag when we pass an element but that's out of scope here. Thanks for clarifying.

Comment (hussain-t) [reply assets/js/components/notifications/ThankWithGoogleSupporterWallNotification.js:56]: I initially used `<p>` tags since, as per the Figma design, we have to render the message as two different paragraphs, not one.

**Review by aaemnnosttv — APPROVED** [review]
Thanks @hussain-t – LGTM 👍

File: assets/js/components/notifications/ThankWithGoogleSupporterWallNotification.js (hunk 3 above)
Comment (aaemnnosttv) [reply assets/js/components/notifications/ThankWithGoogleSupporterWallNotification.js:56]: I opened #5972 to address this.
